numpy/core/fromnumeric.py

```diff
@@ -869,6 +869,11 @@ def sort(a, axis=-1, kind=None, order=None):
         but unspecified fields will still be used, in the order in which
         they come up in the dtype, to break ties.
 
+    by: tuple or object, optional
+        With a given array `a` sort the elements not by arithmetic
+        order, but by any arbitrary order defined by tuple, there can
+        be multiple keys in a tuple form , or just a single object
```

Comment:
I think it would help here to call out what order the keys are sorted in (similar to lexsort documentation). Since trying to follow lexsort it seems like to sort lexicograpphically with keys for complex numbers one should do `arr.sort(by=(c.imag, c.real))`